### DISCUSSION TITLE
cannot find generated types

### DISCUSSION THREAD

**Discussion by sweet-potato-yoo** [prisma/prisma#15999] [Q&A]
Hi, I am using Prisma and PostgreSQL database. My database is filled with some data. I've created initial migration by db pull that creates my database schema. That migration wants to reset my PostgreSQL database. I've tried to avoid it by following this article: https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project. But it didn't actually help or I did something wrong. After baselining, when I am trying to run my server node index.js I am getting an error like:

Unknown type "usersCreateInput"
Unknown type "usersWhereInput".
at assertValidSDL (..\node_modules\graphql\validation\validate.js:135:11)
at buildASTSchema (..\node_modules\graphql\utilities\buildASTSchema.js:44:34)
at makeExecutableSchema (..\node_modules@graphql-tools\schema\cjs\makeExecutableSchema.js:74:47)
at Object. (..\index.js:41:5)
at Module._compile (node:internal/modules/cjs/loader:1155:14)
at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
at Module.load (node:internal/modules/cjs/loader:1033:32)
at Function.Module._load (node:internal/modules/cjs/loader:868:12)
at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
at node:internal/main/run_main_module:22:47

It seems like it cannot find auto-generated graphql input types. How to check if these types are actually auto-generated?
Is it possible to handle this? Thank you in advance)
And why is it wants to reset my PostgreSQL database when I am doing initial migration? Is that means that my prisma schema and PostgreSQL database schema are different? Why can not it just create _prisma_migrations table and add my initial migration into that table?
How can I just create initial migration and _prisma_migrations table, and add migration to this table without losing my data?

**Comment by austincrim** [comment]
Hey @SungHeeYoo 👋,

Could you try the steps outlined [in this comment](https://github.com/prisma/prisma/issues/11555#issuecomment-1027242342) for baselining without a reset?

Regarding the GraphQL types, what library are you using to generate types? Do you have code examples you can share?

**Reply by sweet-potato-yoo** [reply]
@austincrim When I am trying to create .sql file by prisma migrate diff and run it - it returns an error that it cannot find that migration even if I am writing whole path:

>Error: P3017
>The migration init could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: "20201207184859_initial_migration")

**Reply by sweet-potato-yoo** [reply]
@austincrim I just used:

"graphql": "^16.6.0",
"graphql-middleware": "^6.1.33",
"graphql-shield": "^7.6.3",
"graphql-upload": "^16.0.2",
"@prisma/client": "^4.5.0",
"prisma": "^4.5.0"

**Reply by austincrim** [reply]
> When I am trying to create .sql file by prisma migrate diff and run it - it returns an error that it cannot find that migration even if I am writing whole path:
```
mkdir -p prisma/migrations/init
npx prisma migrate diff --preview-feature --from-empty --to-schema-datamodel prisma/schema.prisma --script > prisma/migrations/init/migration.sql
npx prisma migrate resolve --applied init
npx prisma migrate dev # it should say everything is in sync
```
You are running these commands in this order?

Regarding the GraphQL stuff, it would be most helpful if you could provide a minimal, working reproduction repository.

**Reply by sweet-potato-yoo** [reply]
It seems like it's happening when I am trying to use multiple databases. I am working with 2 PostgreSQL databases, so I have 2 prisma folders and 2 generated clients folders. <br>In my schemas I'm generating clients like this:
`generator client {
  provider = "prisma-client-js"
  output = "../generated/client1"
}`. <br> Do you know how to handle with graphql resolvers and typedefs when I am working with multiple databases? Right now my working rtee looks like this:<br>

- generated
-- client1
-- client2
- graphql
-- table1
--- resolvers.js
--- typedefs.js
-- table2
--- resolvers.js
--- typedefs.js
-- resolvers.js
-- typedefs.js
- prisma
-- migrations
-- schema1.prisma
- prisma2
-- migrations
-- schema2.prisma
- index.js

**Reply by austincrim** [reply]
I will need a minimal reproduction repository to dig into this issue.

**Reply by vladi-stevanovic** [reply]
:wave: Hello @SungHeeYoo , I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
